scripts/get_golang_version.sh

```diff
@@ -0,0 +1,2 @@
+#!/usr/bin/env bash
+echo $(cat ./go.mod | grep 'go ' | head -n 1 | cut -d' ' -f2)
```

Comment:
Here are some simpler alternatives:
```
sed -n 's/^go\s\([.0-9]*\)$/\1/p' go.mod

awk '/^go /{ print $0 }' go.mod | cut -d\  -f2
```
There's probably a way to do it with `awk` in one go with group pattern matching where it's not having to use another process like `cut`.

Also, I'm pretty sure that `shellcheck` would have flagged your command above.  Have you installed it and integrated it with your editor?

Reply:
Removed the use of `cat` to satisfy `shellcheck`.

Reply:
Updated to use awk which is cleaner.  Checked with shellcheck.

Reply:
That looks great.

A couple small things.  It's not necessary to capture the output of `awk` in a variable and `echo` that.  Just have the `awk` command output to its caller.

Also, that regex should be tightened, it's a bit "hole-y" right now.  Something like this would make sure that only the desired line is selected:
```
awk '/^go\s[0-9.]+/{print $2}' go.mod
```

Reply:
Updated awk regex to be more specific.

Reply:
The `echo` is unnecessary.  Just have the `awk` command.

Please integrate your editor with `shellcheck` or install a githook.  It will save this kind of back and forth.

While it works in most cases, your regex is overly-complex and does fails some tests.  The one I pasted in above is simple and just uses one character class, matches any number of {number}. pairs, and matches any whitespace (actually, I modified it a bit to match any number of whitespace before the number sequence, we don't really care after it).

```
awk '/^go\s*[0-9.]+/{print $2}' go.mod
```

Reply:
Removed echo.  What tests does the current awk command not support?

Reply:
- It would fail if the tooling changed to place any text after the version string.  I think it would be better to match what we need rather than the whole line, which is what your regex is doing with the `$` anchor.

- It wouldn't catch versions that have four positions, i.e., `1.12.34.56`.

Reply:
This can be updated when and if the tooling changes or go starts supporting 4 levels of version numbers.  Currently, the go version supports major, minor, and optional patch number.  What you are suggesting is more "hole-y".

Reply:
Ok, sounds good.